File: src/main/java/pl/training/module7/breakresponsibilities/before/LegacyDeploymentReport.java
package pl.training.module7.breakresponsibilities.before;

import java.util.List;
import java.util.Objects;

import pl.training.module7.breakresponsibilities.DeploymentSample;

public final class LegacyDeploymentReport {
    public String generate(List<DeploymentSample> samples) {
        Objects.requireNonNull(samples, "samples");

        int deployments = 0;
        int failures = 0;
        long totalLeadTimeMinutes = 0;

        for (DeploymentSample sample : samples) {
            Objects.requireNonNull(sample, "sample");
            deployments++;
            if (!sample.successful()) {
                failures++;
            }
            totalLeadTimeMinutes = Math.addExact(
                    totalLeadTimeMinutes, sample.leadTimeMinutes());
        }

        long averageLeadTimeMinutes = deployments == 0
                ? 0
                : totalLeadTimeMinutes / deployments;

        return "deployments=" + deployments
                + ";failures=" + failures
                + ";avgLeadTimeMinutes=" + averageLeadTimeMinutes;
    }
}

File: typescript/src/module7/breakresponsibilities/before/LegacyDeploymentReport.ts
import { requireNonNull } from '../../../shared/requireNonNull.js';
import { addExact } from '../../ExactMath.js';
import type { DeploymentSample } from '../DeploymentSample.js';

export class LegacyDeploymentReport {
  generate(samples: readonly DeploymentSample[]): string {
    requireNonNull(samples, 'samples');

    let deployments = 0;
    let failures = 0;
    let totalLeadTimeMinutes = 0;

    for (const sample of samples) {
      requireNonNull(sample, 'sample');
      deployments++;
      if (!sample.successful) {
        failures++;
      }
      totalLeadTimeMinutes = addExact(
        totalLeadTimeMinutes, sample.leadTimeMinutes);
    }

    const averageLeadTimeMinutes = deployments === 0
      ? 0
      : Math.trunc(totalLeadTimeMinutes / deployments);

    return 'deployments=' + deployments
      + ';failures=' + failures
      + ';avgLeadTimeMinutes=' + averageLeadTimeMinutes;
  }
}
